src/components/admin/Toggle/Toggle.tsx

```diff
@@ -1,32 +1,54 @@
 import React, { ReactNode, SyntheticEvent, useCallback } from "react";
+import { UseFormRegister } from "react-hook-form";
+
+import { AnyForm } from "types/utility";
 
 import * as TW from "./Toggle.tailwind";
 
 export interface ToggleProps {
   label: ReactNode;
   checked?: boolean;
   onChange?: React.ReactEventHandler<SyntheticEvent>;
+  register?: UseFormRegister<AnyForm>;
   name?: string;
+  title?: string;
 }
 
 export const Toggle: React.FC<ToggleProps> = ({
   label,
   checked = false,
   onChange,
-  name,
+  register,
+  name = "",
+  title,
 }) => {
   // NOTE: the click handlers needs to be on the label to capture all its children clicks and there must always be a <label>
-  const handleClick = useCallback((event) => onChange?.(event), [onChange]);
+  const handleClick = useCallback(
+    (event) => {
+      // default action has to be prevented in order to avoid `double click`
+      onChange && event.preventDefault();
```

Comment:
Please remove this. The Toggle was fixed.

Whatever this "double click" problem is, if it is on the parent side, not inside this component, the parent's `onChange={changeHandler}` can do the `if() return`

On another note, I've already written about why we have avoided `preventDefault()` and this is also a case.